dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitorTest.kt

```diff
@@ -453,6 +452,80 @@ internal class DatadogRumMonitorTest {
             .isEqualTo(viewId)
     }
 
+    @Test
+    fun `stopResource updates relevant view if resource stopped after new view started`(
+        forge: Forge
+    ) {
+        val viewKey = forge.anAlphabeticalString()
+        val viewName = forge.aStringMatching("[a-z]+(\\.[a-z]+)+")
+        val viewKey2 = forge.anAlphabeticalString()
+        val viewName2 = forge.aStringMatching("[a-z]+(\\.[a-z]+)+")
+        val resourceKey = forge.anAlphabeticalString()
+        val resourceMethod = forge.anElementFrom("GET", "PUT", "POST", "DELETE")
+        val resourceUrl = forge.aStringMatching("http(s?)://[a-z]+.com/[a-z]+")
+        val resourceKind = forge.aValueFrom(RumResourceKind::class.java)
+        val attributes = forge.exhaustiveAttributes()
+
+        testedMonitor.startView(viewKey, viewName, emptyMap())
+        val viewId = GlobalRum.getRumContext().viewId
+        testedMonitor.startResource(resourceKey, resourceMethod, resourceUrl, attributes)
+        testedMonitor.startView(viewKey2, viewName2, emptyMap())
```

Comment:
maybe you should also stop this view here to make the test more relevant. It will make it more readable if you will have also the second View event there in the assertion list